CouplingDataUser.C

```diff
@@ -31,11 +31,32 @@ void preciceAdapter::CouplingDataUser::setPatchIDs(std::vector<int> patchIDs)
     patchIDs_ = patchIDs;
 }
 
-void preciceAdapter::CouplingDataUser::setLocationsType(std::string locationsType)
+void preciceAdapter::CouplingDataUser::setLocationsType(LocationType locationsType)
 {
     locationsType_ = locationsType;
 }
 
+void preciceAdapter::CouplingDataUser::checkDataLocation(const bool meshConnectivity)
+{
+    if (this->isLocationTypeSupported(meshConnectivity) == false)
```

Comment:
Why do we need `this->` here? I guess I am overseeing something, but isn't

```suggestion
    if (!isLocationTypeSupported(meshConnectivity))
```
enough?

Reply:
I removed the `this` pointer, but I will not remove the 'equal false' since this is much more readable than the `!`